### PR TITLE
Refine vbar sizing code

### PR DESCRIPTION
**Pull request by taoyuhong**
Simplify the vBAR sizeing code.
1.when update vBAR, always store (value & size_mask | type_bit) to vfcg.
2.when update vBAR base, check if the vBAR value is in address window,
so sizing vBAR will not cause re-mapping base.
3.Remove pci_vdev_read_vbar().

### CODE REVIEW

File: hypervisor/include/dm/vacpi.h [hunk 1]
```diff
@@ -26,6 +26,10 @@
 #define UOS_VIRT_PCI_MMCFG_BASE		0xE0000000UL
 #define UOS_VIRT_PCI_MMCFG_START_BUS	0x0U
 #define UOS_VIRT_PCI_MMCFG_END_BUS	0xFFU
+#define UOS_VIRT_PCI_MEMBASE32      0x80000000UL    /* 2GB */
+#define UOS_VIRT_PCI_MEMLIMIT32     0xC0000000UL    /* 3GB */
```
Comment (lifeix) [inline hypervisor/include/dm/vacpi.h:30]: I think this should change to 0xE0000000UL /**3.5 GB/. Same with the value which is defined in acrn-dm.

**Review by lifeix — CHANGES_REQUESTED** [review]

File: hypervisor/include/dm/vacpi.h (hunk 1 above)
Comment (taoyuhong) [reply hypervisor/include/dm/vacpi.h]: Fixed， change UOS_VIRT_PCI_MEMLIMIT32  to 3.5 GB

File: hypervisor/dm/vpci/vdev.c [hunk 2]
```diff
@@ -137,7 +145,11 @@ static void pci_vdev_update_vbar_base(struct pci_vdev *vdev, uint32_t idx)
 		}
 	}
 
-	/* TODO: 1. check whether the address locate in the MMIO windows 2. base must aligned with size */
+	if ( (base != 0UL) && (!is_pci_io_bar(vbar))) {
+		if ((!is_pci_mem_bar_base_valid(vpci2vm(vdev->vpci), base)) || (!mem_aligned_check(base, vdev->vbars[idx].size))) {
+			base = 0UL;
```
Comment (lifeix) [inline hypervisor/dm/vpci/vdev.c:133]: Hi yuhong

Would you please (1) add error message here; (2) remove the base = 0UL;

**Review by lifeix — CHANGES_REQUESTED** [review]

**Comment by wenlingz** [comment]
OK to verify

**Comment by wenlingz** [comment]
try verify EHL

**Comment by dongyaozu** [comment]
@lifeix can you review?

**Comment by lifeix** [comment]
> @lifeix can you review?

LGTM.

**Review by dongyaozu — APPROVED** [review]

**Pull request merged**